tests/mtouch/MTouch.cs

```diff
@@ -3242,6 +3282,34 @@ public void OnlyDebugFileChange (MTouchLinker linker_options)
 			}
 		}
 
+		[Test]
+		public void MT2003 ()
+		{
+			using (var mtouch = new MTouchTool ()) {
+				mtouch.CreateTemporaryApp ();
+				mtouch.Linker = MTouchLinker.DontLink;
+				mtouch.Debug = true; // makes simlauncher possible, which speeds up the build
+				mtouch.Optimize = new string [] { "all"};
+				mtouch.AssertExecute (MTouchAction.BuildSim);
+				mtouch.AssertWarning (2003, "Option '--optimize=remove-uithread-checks' will be ignored since linking is disabled");
+				mtouch.AssertWarning (2003, "Option '--optimize=dead-code-elimination' will be ignored since linking is disabled");
+				mtouch.AssertWarning (2003, "Option '--optimize=inline-isdirectbinding' will be ignored since linking is disabled");
+				mtouch.AssertWarning (2003, "Option '--optimize=inline-intptr-size' will be ignored since linking is disabled");
+				mtouch.AssertWarning (2003, "Option '--optimize=inline-runtime-arch' will be ignored since linking is disabled");
```

Comment:
we need to make provision for a UI (even if it comes later)
so this should be reflected into the features we publish in our .plist (both XI and XM) and, ideally, the test (maybe not this one) would either use the plist (or make sure it's in sync).

Reply:
@spouliot I added each option to the Versions.plist, with a short description meant to be shown in the UI: f56949bd01dcd01a128521171e9cef1f73efd1ca.

The test will start failing when new options are added, so it won't be possible to not update the test.